mozilla/bedrock
# Remove legacy FxA sync flows (Fixes #9144)

## Description

Bedrock contains some JS logic to help users sign in to their Firefox Account. This includes configuring their Firefox browser for [Sync](https://www.mozilla.org/en-US/firefox/sync/). Since Firefox 80, the Sync configuration logic in bedrock has been handled via [UITour](https://bedrock.readthedocs.io/en/latest/uitour.html).

This PR removes the old Sync logic for Firefox 79 and below, so now only Firefox 80 and above will actively configure Sync for desktop. Older Firefox browsers will simply log in to accounts.firefox.com, in the same way non-Firefox desktop browsers do. 

This PR also converts the files to more modern JS, so apologies if it seems like more to review than it actually is (most of the code is not new, and the files are already well unit tested thankfully).

## Issue / Bugzilla link
#9144

## Testing
The most important thing to test is that the flow still works as expected for modern Firefoxes. Everything should be the same as it is currently in production.

Demo testing prerequisites:
1. Create and open new Firefox profile via `about:profiles` (I tested using current release version FF 95).
2. Open `about:config` and set the following properties/values:
  - `identity.fxaccounts.autoconfig.uri`: `https://accounts.stage.mozaws.net/`
  - `identity.fxaccounts.remote.root`:  `https://accounts.stage.mozaws.net/`
  - `browser.uitour.testingOrigins`: `https://www-demo1.allizom.org/`
2. Restart Firefox (make sure you re-open your testing profile). Verifying the above values look correct after a restart is a good way to confirm things are set correctly.

- [x] Test Fxa Sync Form:
1. Open https://www-demo1.allizom.org/en-US/firefox/browsers/
2. Scroll to the Firefox Accounts form at the bottom of the page.
3. Verify that the form HTML contains hidden input fields with values for `flow_id`, `flow_begin_time`, `device_id`.
4. Enter `success@example.com` in the email field and click "Continue".
5. Verify that you are redirected to https://accounts.stage.mozaws.net with the email address pre-filled.
6. Verify the form says "to continue to Firefox Sync".

- [x] Test Fxa Sync Links:
1. Open https://www-demo1.allizom.org/de/firefox/96.0/whatsnew/?signed-in=false
2. Verify the "Jetzt Konto erstellen" link contains parameters for `flow_id`, `flow_begin_time`, `device_id`.
3. Click the "Jetzt Konto erstellen" link.
4. Verify that you are forwarded to https://accounts.stage.mozaws.net/ with an email signup form.
5. Verify the form says "to continue to Firefox Sync".

- [x] Test FxA product buttons:
1. Open https://www-demo1.allizom.org/en-US/firefox/welcome/1/
2. Verify the "Check Your Breach Report" link contains parameters for `flow_id`, `flow_begin_time`, `device_id`.
3. Click the "Check Your Breach Report" link.
4. Verify you are forwarded to https://accounts.firefox.com/ to sign in to Monitor.

- [x] Test FxA referral params:
1. Open https://www-demo1.allizom.org/en-US/products/vpn/?utm_source=test&utm_campaign=test
2. Scroll down to the pricing section.
3. Verify each of the three subscription links contain `utm_source=test` and `utm_campaign=test`.

- [x] Repeat the above tests but use `https://accounts.firefox.com.cn/` for both `identity.fxaccounts.autoconfig.uri` and `identity.fxaccounts.remote.root` (this test is optional as the logic is all the same).

Wider list of URLs effected by this change:
- https://www-demo1.allizom.org/en-US/firefox/browsers/
- https://www-demo1.allizom.org/en-US/firefox/browsers/mobile/
- https://www-demo1.allizom.org/en-US/firefox/firstrun/
- https://www-demo1.allizom.org/en-US/firefox/
- https://www-demo1.allizom.org/en-US/firefox/97.0a1/whatsnew/
- https://www-demo1.allizom.org/en-US/firefox/pocket/
- https://www-demo1.allizom.org/en-US/firefox/privacy/products/
- https://www-demo1.allizom.org/en-US/firefox/products/
- https://www-demo1.allizom.org/en-US/firefox/sync/
- https://www-demo1.allizom.org/en-US/firefox/welcome/1/
- https://www-demo1.allizom.org/en-US/firefox/welcome/2/
- https://www-demo1.allizom.org/en-US/firefox/welcome/3/
- https://www-demo1.allizom.org/en-US/firefox/accounts/
- https://www-demo1.allizom.org/de/firefox/93.0/whatsnew/
- https://www-demo1.allizom.org/en-US/firefox/93.0/whatsnew/
- https://www-demo1.allizom.org/es-ES/firefox/93.0/whatsnew/
- https://www-demo1.allizom.org/fr/firefox/93.0/whatsnew/
- https://www-demo1.allizom.org/it/firefox/93.0/whatsnew/
- https://www-demo1.allizom.org/nl/firefox/93.0/whatsnew/
- https://www-demo1.allizom.org/de/firefox/95.0/whatsnew/
- https://www-demo1.allizom.org/de/firefox/96.0/whatsnew/
- https://www-demo1.allizom.org/fr/firefox/96.0/whatsnew/
